ls1intum/Artemis
# `Programming exercises`: Make sure template builds are executed when test cases change

### Checklist
#### General


#### Server
#### Changes affecting Programming Exercises

### Motivation and Context
This is a bugfix: the template build is not triggered automatically anymore when an instructor changes the tests

### Description
* I made the minimal change so that the template build will be triggered again. This could trigger the template build in a situation when it is actually not necessarily wanted, i.e. when the instructor changes tests and then manually triggers the solution build, however, this is not really an issues and we better trigger it once too often in an edge case instead of not triggering it at all
* Additionally, I refactored a method to not return an optional anymore, because this complicated the code and its understandability quite a bit, in particular because returning null only happens when an error occurs

### Steps for Testing

Prerequisites:
- 1 Instructor
- 1 Programming Exercise

1. Change something in the test repo and push the changes (either locally or in the online code editor)
2. Wait for the solution build to finish
3. Check that the template build is triggered after the solution build finished
4. Now push changes to the solution repo (without changing the test repo)
5. Make sure only the solution build is triggered
6. Make sure the template build is **not** automatically triggered subsequently


### Review Progress
#### Code Review
#### Manual Tests
